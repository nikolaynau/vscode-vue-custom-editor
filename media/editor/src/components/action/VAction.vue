<script setup lang="ts">
export interface Props {
  tag?: string;
  icon?: string;
  title?: string;
}

withDefaults(defineProps<Props>(), {
  tag: 'div',
  icon: undefined,
  title: undefined
});

const emit = defineEmits<{
  (e: 'click', ev: MouseEvent | KeyboardEvent): void;
}>();

function onClick(e: MouseEvent | KeyboardEvent) {
  emit('click', e);
}
</script>

<template>
  <component
    class="v-action"
    :is="tag"
    :title="title"
    tabindex="0"
    @click="onClick"
    @keyup.space.enter="onClick"
  >
    <span v-if="icon" :class="`codicon codicon-${icon}`" />
  </component>
</template>

<style>
@import 'action';
</style>
