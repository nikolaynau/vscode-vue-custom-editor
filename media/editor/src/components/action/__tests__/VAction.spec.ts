import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import VAction from '../VAction.vue';

describe('VAction', () => {
  it('render default props', () => {
    const wrapper = mount(VAction);
    expect(wrapper.element.tagName).toBe('DIV');
    expect(wrapper.classes('v-action')).toBe(true);
    expect(wrapper.attributes('tabindex')).toBe('0');
    expect(wrapper.attributes('title')).toBeUndefined();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('tag prop', () => {
    const wrapper = mount(VAction, { props: { tag: 'span' } });
    expect(wrapper.element.tagName).toBe('SPAN');
  });

  it('title prop', () => {
    const wrapper = mount(VAction, { props: { title: 'some title' } });
    expect(wrapper.attributes('title')).toBe('some title');
  });

  it('icon prop', () => {
    const wrapper = mount(VAction, { props: { icon: 'clear-all' } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
