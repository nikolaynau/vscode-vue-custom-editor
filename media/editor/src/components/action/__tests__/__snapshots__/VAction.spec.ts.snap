// Vitest Snapshot v1

exports[`VAction > default render props 1`] = `
"<div class=\\"v-action\\" tabindex=\\"0\\">
  <!--v-if-->
</div>"
`;

exports[`VAction > icon prop 1`] = `"<div class=\\"v-action\\" tabindex=\\"0\\"><span class=\\"codicon codicon-clear-all\\"></span></div>"`;
